File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
       
        res=""
        m = 0
        for i in s:
            if i in res:
                res=res[res.index(i)+1:]+i
                

            else:
                res += i
            m = max(m, len(res))
        return (m)

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string str)
    {
        int n = str.size();
 
    int res = 0;
    vector<int>l(256, -1);
    int i = 0;
    for (int j = 0; j < n; j++) {
        i = max(i, l[str[j]] + 1);
        res = max(res, j - i + 1);
        l[str[j]] = j;
    }
    return res;
}
};